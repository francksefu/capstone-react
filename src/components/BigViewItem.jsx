import { PropTypes } from 'prop-types';
import { BsArrowRightCircle } from 'react-icons/bs';
import { useDispatch } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { goWithData } from '../redux/dataSlice/dataSlice';

const BigViewItem = ({ finance }) => {
  const dispatch = useDispatch();
  return (
    <div className="right-flex">
      <NavLink to="/detail" onClick={() => dispatch(goWithData(finance))}>
        <div className="fleche">
          <BsArrowRightCircle />
        </div>
        <h4>{finance.date}</h4>
        <div>
          <span>net debt:</span>
          <br />
          <span>
            {' '}
            $
            {finance.netDebt}
          </span>
        </div>
      </NavLink>
    </div>
  );
};

BigViewItem.propTypes = {
  finance: PropTypes.instanceOf({}).isRequired,
};
export default BigViewItem;
