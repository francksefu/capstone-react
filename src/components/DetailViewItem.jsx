import { useSelector } from 'react-redux';
import { BsArrowRightCircle } from 'react-icons/bs';
import Navbar from './Navbar';

const DetailViewItem = () => {
  const dates = useSelector((store) => store.finance.date);
  if (dates === {}) {
    return <h1>Something went wrong...</h1>;
  }
  return (
    <div>
      <Navbar view="Balance detail" year={' '} />
      <div className="row-40">
        <div className="right-flex">{}</div>
        <div className="left-flex">
          <h2>{dates.date}</h2>
          <p>
            debt:
            {dates.netDebt}
          </p>
        </div>
      </div>
      <h4 className="center">Details</h4>
      <table className="customers">
        <tr>
          <td>cash and cash equivalents : </td>
          <td className="separate">
            <span>
              {dates.cashAndCashEquivalents}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
        <tr>
          <td>short term investments : </td>
          <td className="separate">
            <span>
              {dates.shortTermInvestments}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
        <tr>
          <td>cash and short term investments : </td>
          <td className="separate">
            <span>
              {dates.cashAndShortTermInvestments}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
        <tr>
          <td>net receivables : </td>
          <td className="separate">
            <span>
              {dates.netReceivables}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
        <tr>
          <td>other current assets : </td>
          <td className="separate">
            <span>
              {dates.otherCurrentAssets}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
        <tr>
          <td>total current assets : </td>
          <td className="separate">
            <span>
              {dates.totalCurrentAssets}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
        <tr>
          <td>property plant equipmentNet : </td>
          <td className="separate">
            <span>
              {dates.propertyPlantEquipmentNet}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
        <tr>
          <td>goodwill : </td>
          <td className="separate">
            <span>
              {dates.goodwill}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
        <tr>
          <td>tax sssets : </td>
          <td className="separate">
            <span>
              {dates.taxAssets}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
        <tr>
          <td>total non current assets : </td>
          <td className="separate">
            <span>
              {dates.totalNonCurrentAssets}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
        <tr>
          <td> total assets: </td>
          <td className="separate">
            <span>
              {dates.totalAssets}
              $
            </span>
            <span>
              <BsArrowRightCircle />
            </span>
          </td>
        </tr>
      </table>
    </div>
  );
};

export default DetailViewItem;
