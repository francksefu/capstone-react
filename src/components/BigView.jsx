import { useSelector } from 'react-redux';
import { useState } from 'react';
import BigViewItem from './BigViewItem';
import Navbar from './Navbar';

const BigView = () => {
  const finances = useSelector((store) => store.finance.finance);
  const isLoading = useSelector((store) => store.finance.isLoading);
  const [data, setData] = useState('');

  if (isLoading) {
    return <h1>Loading ... </h1>;
  }

  return (
    <div className="container">
      <Navbar view="Balence view" year={' '} />
      <div className="row-40">
        <div className="right-flex">{}</div>
        <div className="left-flex">
          <h2>Balance Sheet Statement</h2>
          <p>
            items:
            {finances.length}
            <br />
            filter by date
            :
            <input
              type="date"
              id="search"
              placeholder="search"
              onChange={(e) => {
                const regex = new RegExp(e.target.value, 'g');
                setData(finances);
                setData((state) => state.filter((val) => {
                  const str = val.date;
                  return regex.test(str);
                }));
              }}
            />
            <br />
          </p>
        </div>
      </div>
      <h4 className="center">items</h4>
      {data ? (
        <div className="row-50 font-color">
          {data.map((finance) => (
            <BigViewItem key={finance.date} finance={finance} />
          ))}
        </div>
      ) : (
        <div className="row-50 font-color">
          {finances.map((finance) => (
            <BigViewItem key={finance.date} finance={finance} />
          ))}
        </div>
      )}

    </div>
  );
};

export default BigView;
