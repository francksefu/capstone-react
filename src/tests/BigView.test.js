import BigView from '../components/BigView';
import BigViewItem from '../components/BigViewItem';
import DetailViewItem from '../components/DetailViewItem';

test('it renders correclty', () => {
  const component = DetailViewItem;
  expect(component).toMatchSnapshot();
});

test('it renders correclty', () => {
  const component1 = BigViewItem;
  expect(component1).toMatchSnapshot();
});

test('it renders correclty', () => {
  const componentR = BigView;
  expect(componentR).toMatchSnapshot();
});
