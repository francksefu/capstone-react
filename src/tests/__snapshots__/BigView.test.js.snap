// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correclty 1`] = `[Function]`;

exports[`it renders correclty 2`] = `[Function]`;

exports[`it renders correclty 3`] = `[Function]`;
